src/seedu/addressbook/commands/Command.java

```diff
@@ -46,6 +46,15 @@ public CommandResult execute(){
     //Note: it is better to make the execute() method abstract, by replacing the above method with the line below:
     //public abstract CommandResult execute();
 
+    /**
+     * Describes whether a command is able to mutate data.
+     *
+     * @return false by default.
```

Comment:
Having a default implementation for the `isMutating` method helps to reduce the amount of code. However, some may argue that choosing `false` (or `true`) as the default is arbitrary and hurts the intuitiveness of the code. Making the method `abstract` forces all child classes to declare whether it is a mutating command or not, rather than depend on an arbitrary decision made in the parent class.